### ISSUE TITLE
[FEATURE] Ingested data needs to be broken into smaller chunks

### ISSUE THREAD

**Issue by dtaivpp** [opensearch-project/demos#59]
### Is your feature request related to a problem?
At the moment our documents `content` field is too long. If it were to be handed to a language model it would severely limit the amount of context we could pass off to a language model. As a result we should break documents into sub documents with a content length of closer to 100-150 words. Also, there should be a 30% overlap in breaking the documents.  

### What solution would you like?
Here is a bit of an example of what an original document could look like:
```json
{ 
     "url": "/docs/test/",
     "content": "Some large document with way too many words. So long and incredibly detailed. "
}
```

And here is the document split with some overlap: 
```json
{
     "url": "/docs/test/",
     "content": "Some large document with way too many words. "
},
{
     "url": "/docs/test/",
     "content": "too many words. So long and incredibly detailed."
}
```

Note, the metadata should be the same for documents with the content field being the only one chunked into smaller parts. 

### Do you have any additional context?
Haystack has a good example of what this can look like with their document pre-processor. 
https://docs.haystack.deepset.ai/docs/preprocessor

**Comment by LucasWang750** [comment]
can i please take this issue

**Comment by dtaivpp** [comment]
Closed with #69
